haystack/components/retrievers/sentence_window_retrieval.py

```diff
@@ -0,0 +1,124 @@
+    EmbeddingRetriever. One of these retrievers can be used to retrieve documents based on a query and then use this
+    component to get the surrounding documents of the retrieved documents.
+    """
+
+    def __init__(self, document_store: DocumentStore, window_size: int = 3):
+        """
+        Creates a new SentenceWindowRetrieval component.
+
+        :param document_store: The document store to use for retrieving the surrounding documents.
+        :param window_size: The number of surrounding documents to retrieve.
+        """
+        if window_size < 1:
+            raise ValueError("The window_size parameter must be greater than 0.")
+
+        self.window_size = window_size
+        self.document_store = document_store
+
+    @staticmethod
+    def merge_documents_text(documents: List[Document]) -> str:
+        """
+        Merge a list of document text into a single string.
+
+        This functions concatenates the textual content of a list of documents into a single string, eliminating any
+        overlapping content.
+
+        :param documents: List of Documents to merge.
+        """
+        sorted_docs = sorted(documents, key=lambda doc: doc.meta["split_idx_start"])
+        merged_text = ""
+        last_idx_end = 0
+        for doc in sorted_docs:
+            start = doc.meta["split_idx_start"]  # start of the current content
+
+            # if the start of the current content is before the end of the last appended content, adjust it
+            start = max(start, last_idx_end)
+
+            # append the non-overlapping part to the merged text
+            merged_text = merged_text.strip()
+            merged_text += doc.content[start - doc.meta["split_idx_start"] :]  # type: ignore
+
+            # update the last end index
+            last_idx_end = doc.meta["split_idx_start"] + len(doc.content)  # type: ignore
+
+        return merged_text
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        docstore = self.document_store.to_dict()
+        return default_to_dict(self, document_store=docstore, window_size=self.window_size)
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SentenceWindowRetrieval":
+        """
+        Deserializes the component from a dictionary.
+
+        :returns:
+            Deserialized component.
+        """
+        init_params = data.get("init_parameters", {})
+
+        if "document_store" not in init_params:
+            raise DeserializationError("Missing 'document_store' in serialization data")
+        if "type" not in init_params["document_store"]:
+            raise DeserializationError("Missing 'type' in document store's serialization data")
+
+        data["init_parameters"]["document_store"] = InMemoryDocumentStore.from_dict(
```

Comment:
Wait, something is off here? Why do we always create InMemoryDocumentStore and not the proper document store serialized in the type field?

Reply:
ups! nice catch! thanks 👍🏽

Reply:
> Wait, something is off here? Why do we always create InMemoryDocumentStore and not the proper document store serialized in the type field?

I guess you are referring to the `retrievers.in_memory` - those are supposed to work with the `InMemoryDocumentStore`

Reply:
Since we have `document_store: DocumentStore` in the init signature, I thought one could use SWR with any store. If so then when deserializing this component we should recreate the correct DocumentStore, whatever type of DocumentStore  user used.

Reply:
found a fix, kudos to @silvanocerza for the nice tip